### PR TITLE
Feature sync  merge to 'develop'

### PR DESCRIPTION
**Pull request by ArthurQiangLi**
Finished login function
Finished register function
Finished history page data function
Finished sync fish level (score) fucntion
All tested with Postman.

### CODE REVIEW

File: backend_app/urls.py
Comment (marklhj-lee) [inline backend_app/urls.py:1]: Just keep levelup url
remove path("post_sync_level/", views.post_sync_level, name="post_sync_level"),

File: backend_app/views.py
Comment (marklhj-lee) [inline backend_app/views.py:1]: For handling None in post sync level is not necessary, frontend will always send you a value accept that if it is none. Don't have it as level 2. Just delete the handling None. Also dont reject if requesting level is lower than user level. 
If database level is higher, we use that. If phone's level is higher we use that. Also we add 1 on frontend. Dont try to add 1.
After making these changes, change level_up to include post sync level and delete post sync level.

**Review by marklhj-lee — CHANGES_REQUESTED** [review]
Message me on what's app if you are confused.

File: DOCUMENTS/d1_srs_requirements.md
Comment (marklhj-lee) [inline DOCUMENTS/d1_srs_requirements.md:1]: Also edit this once you finish editing post sync level to just level up

File: backend_app/urls.py
Comment (marklhj-lee) [reply backend_app/urls.py:1]: Remove comment.

File: backend_app/views.py
Comment (marklhj-lee) [reply backend_app/views.py:1]: I wouldn't return 403 status code since this is intended behaviour. Remove it please.

**Review by marklhj-lee — CHANGES_REQUESTED** [review]
Change it please.

**Review by marklhj-lee — APPROVED** [review]
Approved

**Comment by ArthurQiangLi** [comment]
Have Fixed the bugs.

**Review by greatfighter — APPROVED** [review]
ok

**Pull request merged**